bin/middlewares/fullText_query.js

```diff
@@ -0,0 +1,71 @@
+const axios = require('axios');
+const yaml = require('node-yaml');
+
+module.exports = function (config) {
+  const idunnBaseUrl = config.server.services.idunn.url || config.services.idunn.url;
+  const geocoderUrl = idunnBaseUrl + '/v1/autocomplete';
+  const useNlu = config.services.geocoder.useNlu;
+  const idunnTimeout = Number(config.server.services.idunn.timeout);
+
+  const categories = yaml.readSync('../../config/categories.yml');
+  const isCategoryValid = type => categories.find(category => category.name === type);
+
+  const removeNullEntries = obj =>
```

Comment:
a `removeNullEntries` function is defined in `url_utils:js`.
Do you think we can reuse this one ?

Reply:
Currently we can't, as it is a client module which can't be required by the server code. I plan to change that soon, but it requires some config change on the server that I didn't want to do in this PR, so I just copied this code knowing it's not ideal.